src/GdbServer.cc

```diff
@@ -900,7 +900,7 @@ bool GdbServer::diverter_process_debugger_requests(
 static bool is_last_thread_exit(const BreakStatus& break_status) {
   // The task set may be empty if the task has already exited.
   return break_status.task_exit &&
-         break_status.task->thread_group()->task_set().size() <= 1;
+         (!break_status.task || break_status.task->thread_group()->task_set().size() <= 1);
```

Comment:
I'm not sure this is entirely correct, as the exited task may not have been the last in its thread group. I'm wondering if either
(a) `BreakStatus` could be extended so `DiversionSession::diversion_step` can specify that the task group really is dead, or
(b) some comments could be sprinkled around to make clear that this is a special case for `DiversionSession` only.

Reply:
Yeah OK I think it makes sense to check `!break_status.task && is_diversion` (which probably means passing in the session here)